upgrade.sh

```diff
@@ -183,14 +139,32 @@ if [ -f tools/solar-only/compose.env ] && [ ! -f ${COMPOSE_ENV_FILE} ]; then
     fi
     rm -rf tz tesla-history weather dashboard.json.bak
     cd ../..
+    echo ""
+fi
 
-    # Add solar-only and weather411 (if configured) to COMPOSE_PROFILES
-    PROFILE="solar-only"
-    add_profile "solar-only"
-    if [ -f weather/weather411.conf ]; then
-        add_profile "weather411"
+# Migrate solar-only profile to pypowerwall cloud mode
+if [ "${PROFILE}" == "solar-only" ]; then
+    # Create Powerwall Settings
+    rm -f ${PW_ENV_FILE}
+    mkdir -p .auth
+    cp tools/tesla-history/tesla-history.auth .auth/.pypowerwall.auth
+    if grep -qE "^SITE = [0-9].+" tools/tesla-history/tesla-history.conf; then
+        grep -E "^SITE = [0-9].+" tools/tesla-history/tesla-history.conf | awk '{ print $NF }' > .auth/.pypowerwall.site
     fi
-    echo ""
+    EMAIL=$(grep -E "^USER = .+@.+" tools/tesla-history/tesla-history.conf | awk '{ print $NF }')
```

Comment:
This migration logic is clever! ! I'll run it across my various OS systems but seems like it would be universal.